src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -300,8 +256,56 @@ set_up(shared_ptr <TargetT > const& target_image_ptr)
 {
 
  if (base_type::set_up(target_image_ptr) == Succeeded::no)
-     error("KOSMAPOSL::set_up(): Error building post filter");
-
+     error("KOSMAPOSL::set_up(): Error setting-up underlying OSMAPOSLReconstruction object");
+
+ if (this->anatomical_image_filenames.empty())
+     error("KOSMAPOSL::set_up(): anatomical_image_filenames has not been set");
```

Comment:
now you prevent using this class without filenames at all. Intentional? Or do you want
```suggestion
 if (this->anatomical_prior_sptrs.empty())
     error("KOSMAPOSL::set_up(): no anatomical prior-info set");
```
On the other hand, with @ashgillman we thought we could allow this size to be 0, such that we can do "PET-only kernel" stuff. That might be going too far here though.

Reply:
mh no I didn't want that actually. maybe the check can be on anatomical_prior==null and hybrid==0

Reply:
I guess `hybrid!=0` then. fine for me (for now certainly)

Reply:
> 
> 
> I guess `hybrid!=0` then. fine for me (for now certainly)

theoretically it should be hybrid!=0 but the iterative part of the kernel is calculated only when hybrid==1
maybe is less confusing to check on sigma_m!=0 if that is zero we can ignore the anatomical part

Reply:
I can't say I understand that one :smile:. For clarity I was thinking you'd want to check
```
if (anatomical_prior==null and hybrid!=0)
  error()
```
A separate check should be (in `set_up`) that sizes are equal. In fact, you could move the whole check there as well I guess.

Tricky to get this all correct of course. I do now think that the `clear()` in `set_defaults()` will be necessary, as otherwise you could have size mismatches if you don't parse.

sorry

Reply:
> 
> 
> I can't say I understand that one 😄. 

Sorry for the confusion. Let's say we would like to run KOSMAPOSL only with the functional (iterative) side information. These estimates are calculated only if hybrid =1. So if we throw an error when (anatomical=null and hybrid!=0) we are not allowing this option. if anatomical=null and hybrid=0, then we would not have any side information.

Reply:
> Tricky to get this all correct of course. I do now think that the `clear()` in `set_defaults()` will be necessary, as otherwise you could have size mismatches if you don't parse.

sorry I don't understand this bit.

Reply:
if `set_defaults()` sets `anatomical_filenames` to a vector of size 1 (albeit with an empty string), but leaves others vectors alone, therefore size 0, it seems that any size checks are going to fail.

Also, if we want to re-use an existing object, one way would be to call `set_defaults` again, so it'd need to get rid of existing arrays. (Although probably we don't expose `set_defaults` at the moment, so this is maybe not anything to worry about).

Reply:
So your suggestion to clear anatomical_filenames (and probably sigma_m) at the beginning of set_default()?

Reply:
yes. all `vector`s that you have really. Hopefully it doesn't make things fall over...

Reply:
all member vectors?

Reply:
Anything that isn't set it in `set_up`, i.e. anything that you think the user should set.